src/wp-includes/option.php

```diff
@@ -3214,3 +3214,126 @@ function wp_autoload_values_to_autoload() {
+ *
+ * @return mixed The network option value for the specified meta key, or false if the function is used incorrectly.
+ */
+function filter_get_site_metadata( $current, $object_id, $meta_key, $single ) {
+	if ( ! is_multisite() ) {
+		_doing_it_wrong( 'get_metadata', __( 'This function is only available in multisite installations.' ), '6.8.0' );
+		return false;
+	}
+
+	if ( ! $single ) {
+		_doing_it_wrong( 'get_metadata', __( 'The $single parameter is not supported in this function.' ), '6.8.0' );
+		return $current;
+	}
+
+	_doing_it_wrong( 'get_metadata', __( 'This function is deprecated. Use get_network_option() instead.' ), '6.8.0' );
+
+	return get_network_option( $object_id, $meta_key );
+}
+
+/**
+ * Filters and updates site metadata in a multisite environment.
+ *
+ * This function is only applicable in multisite installations. It has been
+ * deprecated in favor of using `update_network_option()`.
+ *
+ * @param mixed $current The current metadata value. Unused.
+ * @param int $object_id Site ID for which the metadata is being updated.
+ * @param string $meta_key Metadata key to update.
+ * @param mixed $meta_value Metadata value to update.
+ *
+ * @return bool|void False if the operation is not applicable or invalid,
+ *                   or the result of `update_network_option()` on success.
+ */
+function filter_update_site_metadata( $current, $object_id, $meta_key, $meta_value ) {
+	if ( ! is_multisite() ) {
+		_doing_it_wrong( 'update_metadata', __( 'This function is only available in multisite installations.' ), '6.8.0' );
+		return false;
+	}
+
+	_doing_it_wrong( 'update_metadata', __( 'This function is deprecated. Use update_network_option() instead.' ), '6.8.0' );
+
+	return update_network_option( $object_id, $meta_key, $meta_value );
+}
+
+/**
+ * Filters the addition of site metadata.
+ *
+ * This function is a custom implementation for adding metadata in a multisite environment.
+ * It ensures that the metadata API handles this case correctly.
+ * This function is deprecated and advises using `add_network_option()` instead.
+ * If WordPress is not running in a multisite environment or if the `$unique` parameter is false,
+ * it will return `$current` or false as appropriate.
+ *
+ * @param mixed $current The return value of the original filter function.
+ * @param int $object_id ID of the object (site ID).
+ * @param string $meta_key Metadata key.
+ * @param mixed $meta_value Metadata value. Must be serializable if non-scalar.
+ * @param bool $unique Whether the metadata key should be unique. If false, `$current` is returned.
+ *
+ * @return bool|mixed False on failure, `$current` if the `$unique` parameter is false, or the result of `add_network_option()`.
+ */
+function filter_add_site_metadata( $current, $object_id, $meta_key, $meta_value, $unique ) {
+	if ( ! is_multisite() ) {
+		_doing_it_wrong( 'add_metadata', __( 'This function is only available in multisite installations.' ), '6.8.0' );
+		return false;
+	}
```

Comment:
See above: We can remove this check and call `add_network_option()` as it works outside of Multisite too.

Reply:
They don't bring any benefit IMO, so they hurt more than they help.

It's admittedly a weird situation that the metadata functions with `$object_type = 'site'` touch the same database tables as the network option functions. This should have never been built like that, but we can't change that now without severe backward compatibility breakage. So the reality is that we have both sets of functions, both of which are commonly used WordPress functions.

I like the idea of adding filters to make them work in a more unified way, which is what I'm suggesting overall. If we can make them work, why warn people unnecessarily?